backend/pkg/api/handlers/auth.go

```diff
@@ -253,6 +289,146 @@ func (h *HandlerService) InternalPostLogout(w http.ResponseWriter, r *http.Reque
 	returnOk(w, nil)
 }
 
+func (h *HandlerService) InternalDeleteUser(w http.ResponseWriter, r *http.Request) {
+	user, err := h.getUserBySession(r)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+
+	// TODO allow if user has any subsciptions etc?
+	err = h.dai.RemoveUser(user.Id)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+
+	returnNoContent(w)
+}
+
+func (h *HandlerService) InternalPutUserEmail(w http.ResponseWriter, r *http.Request) {
+	// validate user
+	user, err := h.getUserBySession(r)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	userInfo, err := h.dai.GetUserInfo(user.Id)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	if !userInfo.Confirmed {
+		handleErr(w, errors.New("cannot update email for an unconfirmed address"))
```

Comment:
We should probably allow changing the email here, e.g. if the user has used a wrong email during signup. Otherwise, this line would cause a 500 to be returned, should rather be 409.

Reply:
I took this from v1 behaviour: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/handlers/user.go#L1300
> We should probably allow changing the email here, e.g. if the user has used a wrong email during signup.

Typically you can't even login to web pages unless your email is confirmed though, right? So I'd rather stick to that, that's also how we did it in v1 (https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/handlers/auth.go#L278).

That's also not there yet, I'll add it for now; LMK if you think we should deviate from this common/v1 behaviour anyways

Reply:
You have convinced me